lib/IDE/CodeCompletion.cpp

```diff
@@ -1037,8 +1035,13 @@ void CodeCompletionCallbacksImpl::addKeywords(CodeCompletionResultSink &Sink,
 
     break;
   case CompletionKind::CaseStmtBeginning:
-  case CompletionKind::TypeIdentifierWithDot:
-  case CompletionKind::TypeIdentifierWithoutDot:
+  case CompletionKind::TypeSimpleWithDot:
+    break;
+
+  case CompletionKind::TypeSimpleWithoutDot:
+    if (isa_and_nonnull<TupleTypeRepr>(ParsedTypeLoc.getTypeRepr())) {
+      addEffectsSpecifierKeywords();
+    }
```

Comment:
Nice. Since it's so super clear at a glance, it'd be great to add comment what this is for.

Reply:
😅 Point taken.